Sources/Commands/SwiftPackageTool.swift

```diff
@@ -966,3 +1051,207 @@ fileprivate func logPackageChanges(changes: [(PackageReference, Workspace.Packag
+    case .executable:
+        packageType = .executable
+    case .library:
+        packageType = .library
+    case .systemModule:
+        packageType = .systemModule
+    case .empty:
+        packageType = .empty
+    case .manifest:
+        packageType = .manifest
+    default:
+        packageType = .library
+    }
+    
+    switch useNewBehaviour {
+    case .new:
+        return PackageTemplate(directories: defaultDir, type: packageType)
+    case .legacy:
+        let legacyDir = PackageTemplate.Directories(sources: RelativePath("./Sources"), tests: RelativePath("./Tests"), createSubDirectoryForModule: true)
+        return PackageTemplate(directories: legacyDir, type: packageType)
+    }
+}
+
+fileprivate func makePackage(fileSystem: FileSystem,
+                             configPath: AbsolutePath,
+                             packageName: String?,
+                             mode: MakePackageMode,
+                             packageType: InitPackage.PackageType?,
+                             packageTemplate: String?) throws {
+    
+    guard let cwd = fileSystem.currentWorkingDirectory else {
+        throw InternalError("Could not find the current working directroy.")
+    }
+    
+    guard !(packageType != nil && packageTemplate != nil) else {
+        throw InternalError("Can't use --type in conjunction with --template")
+    }
+    
+    let name: String
+    let destinationPath: AbsolutePath
+    let templateHomeDirectory = configPath.appending(components: "templates", "new-package")
+    
+    var foundTemplate = false
+    var templateToUse = ""
+    
+    if let templateName = packageTemplate {
+        guard fileSystem.exists(templateHomeDirectory.appending(component: templateName)) else {
+            throw InternalError("Could not find template folder: \(templateHomeDirectory.appending(component: templateName))")
+        }
+        
+        templateToUse = templateName
+        foundTemplate = true
+    } else {
+        // Checking if a default template is present
+        if fileSystem.exists(templateHomeDirectory.appending(components: "default")) {
+            templateToUse = "default"
+            foundTemplate = true
+        }
+    }
+    
+    switch mode {
+    case .initialize:
+        name = packageName ?? cwd.basename
+        destinationPath = cwd
+    case .create:
+        name = packageName!
+        try localFileSystem.createDirectory(cwd.appending(component: name))
+        destinationPath = cwd.appending(component: name)
+    }
+    
+    let packageTemplate: InitPackage.PackageTemplate
+    if foundTemplate && useNewBehaviour == .new {
+        try fileSystem.getDirectoryContents(templateHomeDirectory.appending(component: templateToUse)).forEach {
+            print("Copying \($0)")
+            try copyTemplate(fileSystem: fileSystem, sourcePath: templateHomeDirectory.appending(components: templateToUse, $0), destinationPath: destinationPath, name: name)
+        }
+        return
```

Comment:
if returning here, then the else branch is not required and you can move the  let packageTemplate down